riptide-idempotency/src/main/java/org/zalando/riptide/idempotency/IdempotencyPredicate.java

```diff
@@ -27,20 +30,27 @@
 
     public IdempotencyPredicate() {
         this(Arrays.asList(
-                new DefaultIdempotencyDetector(),
                 new ConditionalIdempotencyDetector(),
+                new DefaultIdempotencyDetector(),
+                new ExplicitIdempotencyDetector(),
                 new IdempotencyKeyIdempotencyDetector(),
                 new MethodOverrideIdempotencyDetector()
         ));
     }
 
     @Override
     public boolean test(final RequestArguments arguments) {
-        return arguments.getAttribute(IDEMPOTENT).orElseGet(() ->
-            detectors.stream()
-                .anyMatch(detector -> detector.test(arguments, modified ->
-                        !arguments.equals(modified) && test(modified)))
-        );
+        return decide(arguments) == Decision.ACCEPT;
+    }
+
+    private Decision decide(final RequestArguments arguments) {
+        final Test test = modified ->
+                arguments.equals(modified) ? NEUTRAL : decide(modified);
+
+        return detectors.stream()
+                .map(detector -> detector.test(arguments, test))
+                .max(naturalOrder())
```

Comment:
Ha trick 17. And now try to also do early exit on `DENY` :wink:

Reply:
Yeah, going to change that to a for-loop with an early break.